src/specialized/aarch64.rs

```diff
@@ -0,0 +1,132 @@
+//use std::arch::aarch64 as arch;
+use stdsimd::arch::aarch64 as arch;
+
+#[derive(Clone)]
+pub struct State {
+    state: u32,
+}
+
+impl State {
+    pub fn new() -> Option<Self> {
+        if is_aarch64_feature_detected!("crc") {
+            // SAFETY: The conditions above ensure that all
+            //         required instructions are supported by the CPU.
+            Some(Self { state: 0 })
+        } else {
+            None
+        }
+    }
+
+    pub fn update(&mut self, buf: &[u8]) {
+        // SAFETY: The `State::new` constructor ensures that all
+        //         required instructions are supported by the CPU.
+        self.state = unsafe { calculate(self.state, buf) }
+    }
+
+    pub fn finalize(self) -> u32 {
+        self.state
+    }
+
+    pub fn reset(&mut self) {
+        self.state = 0;
+    }
+
+    pub fn combine(&mut self, other: u32, amount: u64) {
+        self.state = ::combine::combine(self.state, other, amount);
+    }
+}
+
+pub unsafe fn calculate(crc: u32, data: &[u8]) -> u32 {
+    let mut len = data.len();
+    let mut c32 = !crc;
+    let mut ptr = data.as_ptr();
+    let mut ptr2;
+    let mut ptr4;
+    let mut ptr8;
+
+    if len != 0 && ((ptr as usize) & 1) != 0 {
```

Comment:
Could this perhaps use the recently stabilized `align_to` method on slices to do the workhorse of the logic around alignment here?

Reply:
oh, it just wasn't inlining the intrinsics' wrappers because I removed the `target_feature` attr. lol.